### PR TITLE
Article/scaling rails 3.old

### PR DESCRIPTION
**Pull request by michael-groble**
Describes what we've done to keep going strong on Rails 3.old

### CODE REVIEW

File: _posts/2017-03-13-scaling-rails-3.old.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+layout: post
+title:  "Scaling Rails 3.old"
+author: "Michael Groble"
+excerpt: >
+  We're not particularly proud that we are still using Rails 3.2, but we _are_ extremely
+  proud of scaling our traffic over the last few years. Our new services are built in Rails 5,
+  but we still have a hefty chunk of funtionality in our original Rails 3.x monolith. A rueful
+  "rails 3.old" hits our slack channels as someone stumbles across an issue as they switch back
+  to the monolith. Growing our business, features and team over the last few years has left
+  little time to do a wholesale upgrade of the monolith platform.  Here are some tactial things
```
Comment (halfwolf) [inline _posts/2017-03-13-scaling-rails-3.old.md]: tactical*

File: _posts/2017-03-13-scaling-rails-3.old.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+```
+
+That is an important scenario for us and we've been running a configuration like that for months now. After flipping
+from Unicorn to Puma, we saw both CPU usage and MGET command counts go way down on our Redis cache instances.  Sure enough,
+the support for those pool parameters aren't in the version that works with Rails 3.  The dependency nightmare that exists in
+all the little redis-store gems means that we will likely need to fork and backport this pool support to move forward with Puma
+or add `Redis::Distributed` support to [readthis](https://github.com/sorentwo/readthis).
+
+## Path Forward
+The good news is that we are on the lastest versions of both Postgres and Redis in production so the Rails stack itself is
```
Comment (halfwolf) [inline _posts/2017-03-13-scaling-rails-3.old.md]: latest*

File: _posts/2017-03-13-scaling-rails-3.old.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+we have been on for months now and it is accelerating as we get those first services running in production.
+
+Tracking down the last Redis cache pool gotcha is what finally prompted me to write a blog post about our experiences.  If
+Rails 3.old is popping up in your slack channels as well, hopefully you find some value in our forks or let us know of
+alternatives you use to keep things running.  Maintaining an old code base is always an adventure.
+
+And if you're interested in working with us and have actually migrated a monolith off of Rails 3, don't think you need to
+hide that on your resume.  We promise we won't lock you away to work on that for your first weeks.  We spread infrastructure
+and platform work across all teams and have way too many interesting things going on to monopolize your time like that (although you
+_may_ get bounus points if your code flags invalid option usage)!
```
Comment (halfwolf) [inline _posts/2017-03-13-scaling-rails-3.old.md]: bonus*

**Comment by gregbeech** [comment]
Nice, really good post!

**Comment by xarisd** [comment]
Really informative blog post 👍 
Works for us new-starters too 😉

**Comment by mezis** [comment]
@jp: Wikipedia has plenty of photos of Dany Devito with appropriate license, e.g. [this one](https://en.wikipedia.org/wiki/Danny_DeVito#/media/File:Danny_DeVito_by_Gage_Skidmore.jpg).

**Comment by michael-groble** [comment]
now with updated pic @jphastings

**Pull request merged**